internal/terraform/evaluate.go

```diff
@@ -268,11 +268,27 @@ func (d *evaluationStateData) GetInputVariable(addr addrs.InputVariable, rng tfd
 	}
 
 	val, isSet := vals[addr.Name]
-	if !isSet {
-		if config.Default != cty.NilVal {
-			return config.Default, diags
-		}
-		return cty.UnknownVal(config.Type), diags
+	switch {
+	case !isSet:
+		// The config loader will ensure there is a default if the value is not
+		// set at all.
```

Comment:
I'm a bit confused after reading this comment. Where does the config loader ensure that a variable has a `Default` set?

I was expecting a diag here if `nullable` is `false` and there's no `Default`. Is that handled elsewhere?

Reply:
This case is assured because in the absence of a default value, a module call must contain an assignment for the variable, therefore `if !isSet` then we must have a default. The comment is only there because we're not checking `config.Default != cty.NilVal`.

Reply:
Oh, and the case of if nullable is false and there's no Default is handled by the 3rd case in this switch

```
case val.IsNull() && !config.Nullable:
```

Why which time we have also determined from the prior case statements that there is an input value set and there is no default.